app/src/sharedTest/java/org/oppia/android/app/onboarding/OnboardingFragmentTest.kt

```diff
@@ -636,6 +674,68 @@ class OnboardingFragmentTest {
     }
   }
 
+  @Test
+  fun testOnboardingFragment_checkSlide3_policiesLinkIsVisible() {
+    launch(OnboardingActivity::class.java).use {
+      onView(withId(R.id.skip_text_view)).perform(click())
+      testCoroutineDispatchers.runCurrent()
+      onView(withId(R.id.slide_terms_of_service_and_privacy_policy_links_text_view)).perform(
+        scrollTo()
+      )
+      onView(withId(R.id.slide_terms_of_service_and_privacy_policy_links_text_view)).check(
+        matches(
+          isDisplayed()
+        )
+      )
+    }
+  }
+
+  @Test
+  fun testHtmlContent_withPolicies_linkSupport_clickSpan_callsTagListener() {
+    val htmlParser = htmlParserFactory.create(
+      policyOppiaTagActionListener = mockPolicyOppiaTagActionListener
+    )
+    testCoroutineDispatchers.runCurrent()
+
+    launch(OnboardingActivity::class.java).use {
+      onView(withId(R.id.skip_text_view)).perform(click())
+      testCoroutineDispatchers.runCurrent()
+
+      val textView: TextView =
+        it.findViewById(R.id.slide_terms_of_service_and_privacy_policy_links_text_view)
+
+      val htmlResult: Spannable = htmlParser.parseOppiaHtml(
+        "By using %s, you agree to our <br> <oppia-noninteractive-policy link=\"tos\">" +
+          " Terms of Service </oppia-noninteractive-policy> and <oppia-noninteractive-policy " +
+          "link=\"privacy\">Privacy Policy </oppia-noninteractive-policy>.",
+        textView,
+        supportsLinks = true,
+        supportsConceptCards = false
+      )
+      textView.text = htmlResult
+
+      // Verify the displayed text is correct & has a clickable span.
+      val clickableSpans = htmlResult.getSpansFromWholeString(ClickableSpan::class)
+      assertThat(htmlResult.toString()).isEqualTo(
+        "By using %s, you agree to our \n" +
+          "Terms of Service and Privacy Policy."
+      )
+      assertThat(clickableSpans).hasLength(2)
+      clickableSpans[0].onClick(textView)
+      clickableSpans[1].onClick(textView)
+
+      // Verify that the tag listener is called.
+      verify(mockPolicyOppiaTagActionListener, times(2)).onPolicyPageLinkClicked(
+        capture(policyPageCaptor)
+      )
+
+      assertThat(policyPageCaptor.allValues).containsExactly(
+        PolicyPage.TERMS_OF_SERVICE,
+        PolicyPage.PRIVACY_POLICY
+      ).inOrder()
+    }
```

Comment:
Prefer splitting this up into 2 tests dedicated to each the privacy policy and terms of service.

Reply:
Done.